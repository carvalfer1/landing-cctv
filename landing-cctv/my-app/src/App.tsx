import React from 'react';
import './design/styles.scss';
import { BrowserRouter as Router, Route } from "react-router-dom";
import Layout from './components/common/Layout';
import Home from './components/home';
import Jest from './components/jest';
import History from './components/history';

class App extends React.Component<{},{}>
{
  constructor(props:any){
    super(props);
  }
  
  render:any = () => {
    return    <Router>
                <Layout>
                  <Route path="/" exact component={Home} />
                  <Route path="/jest" component={Jest} />
                  <Route path="/history" component={History} />
                </Layout>
              </Router>
            
  }  
}

export default App;