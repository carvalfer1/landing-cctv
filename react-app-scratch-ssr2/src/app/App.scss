@import "./design/params.scss";
@import "./design//colors.scss";

$h4fontsize: 16px;
$h4fontweight: 600;
$h3fontsize: 30px;
$h3fontweight: 500;
.application {
  .icon-down {
    margin-top: 30px;
    font-size: 30px;
    color: #ffffff75;
    cursor: pointer;
  }
  h1,
  h3,
  h4 {
    font-weight: 200 !important;
  }
  h3 {
    font-size: 20px;
  }
  .header-container-text{
    display: flex;
    flex-direction: column;
    text-align: left;
    span{
      font-weight: 200;
      font-size: 1.4rem !important;
      margin-bottom: 5px;
      letter-spacing: 1px;
    }
  }
  .header {
    display: block;
    .header-layer-container {
      background: black;
      max-height: $content-width;
      overflow: hidden;
      .header-layer-img {
        opacity: 0.5;
        width: 100%;
        filter: blur(4px);
        /** filter: blur(4px); */
        /** -webkit-filter: blur(4px); */
      }
    }
    .top-container {
      position: absolute;
      top: 10px;
      text-align: center;
      width: 100%;
      display: flex;
      justify-content: center;

      > div {
        width: 100%;
        max-width: 900px;
        display: flex;
        align-items: center;
        img {
          width: 270px;
        }
        div {
          &:first-child {
            text-align: left;
          }
        }
      }

      .call-us {
        text-align: right;
        margin-top: 4px;
        font-size: 14px;
      }

      span {
        margin-left: 4px;

        color: white;
      }
    }

    .central-title {
      position: absolute;
      width: 100%;
      text-align: center;
      display: flex;
      /* align-items: center; */
      justify-content: center;

      h1 {
        color: white;
      }
      span {
        color: white;
      }
      div {
        max-width: $content-width;
      }
    }
  }

  .ant-card {
    max-height: 200px;
    min-height: 150px;
    margin-bottom: 12px;
  }
  .ant-card-bordered {
    margin-right: 12px !important;
  }
  .ant-card-head-title {
    button {
      cursor: pointer;
    }
  }

  .cameras {
    display: flex;
    justify-content: center;
    img {
      position: absolute;
    }
  }

  .block-1 {
    padding-left: 20px;
    padding-right: 20px;
    display: flex;
    justify-content: center;
    div {
      max-width: $content-width;
      margin-top: 200px;
      text-align: center;
      margin-bottom: 40px;
      h4 {
        margin-bottom: 0px;
        font-size: $h4fontsize;
        font-weight: $h4fontweight;
      }
      h1 {
        font-size: $h3fontsize;
        font-weight: $h3fontweight;
      }
    }
  }

  .block-2 {
    justify-content: center;
    display: flex;
    text-align: center;
    padding: 60px 20px;
    background: #262626;
    > div {
      max-width: $content-width;
      .features {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        div {
          padding: 0px 10px;
          width: 280px;
          i {
            font-size: 40px;
            margin-top: 22px;
            margin-bottom: 10px;
            color: #15808f;
          }
        }
      }
      h3,
      h4,
      span {
        color: white;
      }
      h4 {
        font-size: $h4fontsize;
        font-weight: $h4fontweight;
      }
      h3 {
        font-size: $h3fontsize;
        font-weight: $h3fontweight;
      }
    }
  }

  .block-3 {
    h4 {
      text-align: center;
    }
    padding: 60px 20px;
    justify-content: center;
    display: flex;
    > div {
      h3 {
        text-align: center;
        font-size: $h3fontsize;
        font-weight: $h3fontweight;
        margin-bottom: 30px;
      }
      max-width: $content-width;
    }
  }

  .block-4 {
    padding: 60px 20px;
    justify-content: center;
    display: flex;
    background: #dadada;
    > div {
      max-width: $content-width;
      h3 {
        text-align: center;
        font-size: $h3fontsize;
        font-weight: $h3fontweight;
        margin-bottom: 30px;
        color: #262626;
      }
      > div {
        .cont {
          display: flex;
          font-size: 14px;
          margin-bottom: 20px;
          color: #5e5e5e;
        }
        img {
          height: 20px;
          width: 20px;
          margin-right: 10px;
        }
        color: white;
        font-size: 14px;
      }
      /*li:before{
                content:''; 
                display:inline-block; 
                height:1em; 
                width:1em; 
                background-image:url('../../assets/check-square-solid.svg'); 
                background-size:contain; 
                background-repeat:no-repeat; 
                padding-left: 2em; 
            }*/
    }
  }

  .block {
    padding: 60px 20px;
    justify-content: center;
    display: flex;
    > div {
      max-width: $content-width;
      h3 {
        text-align: center;
        font-size: $h3fontsize;
        font-weight: $h3fontweight;
        margin-bottom: 30px;
        color: #262626;
      }
    }
  }

  .instalation {
    background-color: #dadada;
    h1 {
      font-size: $h3fontsize;
      text-align: center;
    }
    .precio {
      flex-direction: column;
      h4 {
        margin: 0px;
      }
    }
    > div {
      h3 {
        margin-bottom: 5px;
      }
      h4 {
        text-align: center;
        margin-bottom: 25px;
      }
      > div {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;
        div {
          min-width: 150px;
          display: flex;
          padding: 20px 40px;
          flex-direction: column;
        }
        .box-instalation {
          border: 1px solid #15808f;
          border-radius: 5px;
          margin: 12px;
          width: 400px;
          h3 {
            color: #15808f;
          }
        }
        .box-client {
          border: 1px solid #15808f;
          border-radius: 5px;
          margin: 12px;
          background: white;
          display: flex;
          justify-content: center;
          h3 {
            color: #15808f;
          }
          img{
            width: 110px;
          }
        }        
        img {
          width: 160px;
        }
      }
    }
    .biter-name {
      text-align: center;
      margin-top: 10px;
      font-size: 16px;
    }
    .biter-range {
      color: #c20a66;
      text-align: center;
      margin-bottom: 10px;
    }
  }

  .promotion {
    border-bottom: 1px solid #dadada;
    border-style: dashed;
    border-left: 0px;
    border-right: 0px;
    border-top: 0px;
    background-color: white;
    > div {
      h2 {
        margin-bottom: 5px;
        font-size: 30px;
        font-weight: 100;
        text-align: center;
      }
      h4 {
        text-align: center;
        margin-bottom: 25px;
      }
      > div {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;
        div {
          min-width: 150px;
          display: flex;
          justify-content: center;
          padding: 20px 40px;
          flex-direction: column;
        }
        .box-promotion {
          border: 1px solid #dadada;
          border-radius: 5px;
          margin: 12px;
          width: 400px;
          .before {
            font-size: 16px;
            text-decoration: line-through;
          }
          .after {
            color: #c20a66;
          }
          .price-discount {
            margin-left: -57px;
            border: 1px solid #dadada;
            width: 65px;
            height: 65px;
            padding-top: 16px;
            border-radius: 50%;
            padding-left: 7px;
            font-size: 20px;
            /* border-style: dashed; */
            background: #c20a66;
            color: white;
          }
        }
        img {
          width: 160px;
        }
      }
    }
    .biter-name {
      text-align: center;
      margin-top: 10px;
      font-size: 16px;
    }
    .biter-range {
      color: #c20a66;
      text-align: center;
      margin-bottom: 10px;
    }

    .valid {
      font-size: 24px;
      margin-top: 25px;
      color: #15808f;
    }
  }

  .block-5 {
    display: none;
    background-color: white;
    > div {
      h3 {
        margin-bottom: 5px;
      }
      h4 {
        text-align: center;
        margin-bottom: 25px;
      }
      > div {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;
        div {
          min-width: 200px;
          display: flex;
          justify-content: center;
          padding: 20px 40px;
          flex-direction: column;
        }
        img {
          width: 160px;
        }
      }
    }
    .biter-name {
      text-align: center;
      margin-top: 10px;
      font-size: 16px;
    }
    .biter-range {
      color: #c20a66;
      text-align: center;
      margin-bottom: 10px;
    }
  }

  .header-mobile {
    .call-us {
      padding-top: 10px;
      padding-bottom: 10px;
      background: #c20a66;
      color: #ffffff;
      font-size: 17px;
      text-align: center;
    }
    display: none;
    > div {
      background-color: $background-gray;
    //   padding: 0px 20px 40px 20px;
      text-align: center;
      > div {
        display: flex;
        justify-content: center;
        img {
          width: 75%;
          max-width: 250px;
        }
      }
    }
  }
}

.footer-home {
  width: 100%;
  display: flex;
  justify-content: center;
  background: #262626;
  .container-max-size {
    padding: 0px 20px;
  }
  > div {
    width: 100%;
    max-width: 900px;
    .bit-information {
      display: flex;
      align-items: center;
      .logo {
        width: 50%;
        text-align: left;

        img {
          width: 300px;
        }
      }
      .contact-information {
        width: 50%;
        div {
          text-align: right;
        }
        color: #5e5e5e;
      }
    }
    .copy {
      text-align: center;
      color: #5e5e5e;
      margin-bottom: 10px;
    }
  }
  .principal-color {
    color: #c20a66;
  }
}
