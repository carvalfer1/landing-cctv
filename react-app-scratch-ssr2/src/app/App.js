import React, { Component } from "react";
import "./App.scss";
import "antd/dist/antd.css";
import "./design/styles.scss";
import visionAltaImg from "../assets/visionalta.jpeg";
import clienteFeliz2 from "../assets/clientefeliz2.jpg";

import { Button, Col, Card, Steps, Icon } from "antd";

import HeaderApp from "./components/header/header";
import FooterApp from "./components/footer/footer";
import Home from "./containers/home/home";
import CovidComponent from "./containers/covid/covid";
import CustomSolutions from "./containers/customSolutions/customSolutions";
import CommunitySolutions from "./containers/communitySolutions/communitySolutions";
import FacialTerminal from "./containers/covid/facialTerminal";
import Blog from "./containers/blog/blog";

import { Router, Route } from "react-router-dom";
import { createMemoryHistory } from "history";
import { bind } from "file-loader";

const history = createMemoryHistory();
// const { pathname, protocol, host } = window.location;

console.log("NODE_ENV: ", process.env.NODE_ENV);

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      headerInfo: {
        background: visionAltaImg,
        headerTitle: "Sistemas de cámaras de seguridad",
        text: (
          <div className="header-container-text">
            <span>
              Protege tu negicio
            </span>
            <span>
              Visualiza tu negicio por internet
            </span>
            <span>
              Desde teléfonos o computadores
            </span>
            <span>
              Instalaciones robustas y duraderas
            </span>
            <span>
              Alta resolución y equipos profesionales        
            </span>
            <span>
              Garantía por instalación y equipos
            </span>
          </div>
        ),
        headerMobile: "Sistemas de cámaras de seguridad",
        textMobile: (
          <div className="header-container-text">
            <span>
              Protege tu negicio
            </span>
            <span>
              Visualiza tu negicio por internet
            </span>
            <span>
              Desde teléfonos o computadores
            </span>
            <span>
              Instalaciones robustas y duraderas
            </span>
            <span>
              Alta resolución y equipos profesionales        
            </span>
            <span>
              Garantía por instalación y equipos
            </span>
          </div>
        ),
        backgroundMobile: clienteFeliz2,
      },
    };
    this.changeImg = this.changeImg.bind(this);
  }

  changeImg(_headerInfo) {
    this.setState({ headerInfo: _headerInfo });
  }

  render() {
    let pathname = null;
    if (this.props.request) {
      console.log("from server");
      console.log(this.props.request);
      pathname = this.props.request;
    } else {
      pathname = window.location.pathname;
    }

    let comp;

    switch (pathname) {
      case "/covid":
        comp = <CovidComponent callBackChangeImg={this.changeImg} />;
        break;
      case "/sistemas-a-medida":
        comp = <CustomSolutions callBackChangeImg={this.changeImg} />;
        break;
      case "/soluciones-comunitarias":
        comp = <CommunitySolutions callBackChangeImg={this.changeImg} />;
        break;
      case "/terminal-facial":
        comp = <FacialTerminal callBackChangeImg={this.changeImg} />;
        break;
      case "/blog":
        comp = <Blog callBackChangeImg={this.changeImg} />;
        break;
      default:
        comp = <Home />;
        break;
    }

    return (
      <Router history={history}>
        <div className="application">
          <HeaderApp {...this.state.headerInfo}></HeaderApp>
          {comp}
          <FooterApp></FooterApp>
        </div>
      </Router>
    );
  }
}

export default App;
