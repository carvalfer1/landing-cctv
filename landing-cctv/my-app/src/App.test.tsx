import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

// only verify that the app or the component render without crashing but
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.unmountComponentAtNode(div);
  // ReactDOM.render(<App/>,div);
});

test('App Born Snaptshot', () => {
  const comp = Enzyme.shallow(<App><div>Content</div></App>);
  // expect(toJson(comp)).toMatchSnapshot(); // we can do at this, but i put a line code in the package.json that automatic this line -> "snapshotSerializers": ["enzyme-to-json/serializer"]
  expect(comp).toMatchSnapshot();
});