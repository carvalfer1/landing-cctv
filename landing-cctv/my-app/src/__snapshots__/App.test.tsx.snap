// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Born Snaptshot 1`] = `
<Layout>
  <div>
    Content
  </div>
</Layout>
`;
